### PR TITLE
feat: Add GitHub App creds support

### PR DESCRIPTION
**Pull request by daniel-codefresh**
Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by daniel-codefresh** [comment]
Related issue: #1372

File: eventsources/sources/github/start.go [hunk 1]
```diff
@@ -52,17 +52,60 @@ func init() {
 	go webhook.ProcessRouteStatus(controller)
 }
 
-// getCredentials for retrieves credentials for GitHub connection
+// getCredentials retrieves credentials for GitHub connection
 func (router *Router) getCredentials(keySelector *corev1.SecretKeySelector) (*cred, error) {
 	token, err := common.GetSecretFromVolume(keySelector)
 	if err != nil {
-		return nil, errors.Wrap(err, "token not founnd")
+		return nil, errors.Wrap(err, "secret not found")
```
Comment (whynowy) [inline eventsources/sources/github/start.go:59]: 👍

**Review by whynowy — APPROVED** [review]
LGTM, as good as always!

**Comment by jmeridth** [comment]
When will this be included in a release? /cc @whynowy

**Pull request merged**